dart-common/src/main/java/com/f2prateek/dart/common/AbstractDartProcessor.java

```diff
@@ -163,6 +167,73 @@ protected InjectionTarget getOrCreateTargetClass(Map<TypeElement, InjectionTarge
     return injectionTarget;
   }
 
+  protected void findParentInjectionTarget(TypeElement typeElement,
+                                                                InjectionTarget injectionTarget) {
+    findInjectionTargetForSuperTypeWithInjectExtra(typeElement,
+            injectionTarget,
+            false,
+            true);
+  }
+
+  protected void findAllParentsInjectionMap(TypeElement typeElement,
+                                            InjectionTarget injectionTarget) {
+    findInjectionTargetForSuperTypeWithInjectExtra(typeElement,
+            injectionTarget,
+            true,
+            false);
+  }
+
+  private void findInjectionTargetForSuperTypeWithInjectExtra(TypeElement typeElement,
+                                                                InjectionTarget injectionTarget,
+                                                                boolean shouldAddInjectionMaps,
+                                                                boolean shouldStopAtFirstParent) {
+    while (true) {
+      TypeMirror superType = typeElement.getSuperclass();
+      if (superType.getKind() == TypeKind.NONE) {
+        return;
+      }
+      final TypeElement superTypeElement = (TypeElement) ((DeclaredType) superType).asElement();
+      final InjectionTarget superInjectionTarget = scanTypeForInjectExtras(superTypeElement);
+      if (!superInjectionTarget.injectionMap.isEmpty()) {
+        if (injectionTarget.parentTarget == null) {
```

Comment:
the second time we come here, `injectionTarget.parentTarger` is already different than null, so we won't inherit the `injectionMap` from older ancestors.

Reply:
Yes, the end bracket should on line 202.